mrtt-ui/src/library/organizeMangroveSpeciesList.js

```diff
@@ -0,0 +1,36 @@
+import { mangroveSpeciesPerCountryList } from '../data/mangroveSpeciesPerCountry'
+
+// mangroveSpeciesPresent list should display country specific species at the top, with all
+// species below the country specific list, removing duplicates from the second list
+
+const organizeMangroveSpeciesList = (siteCountriesResponse) => {
+  const countriesList = siteCountriesResponse.map((countryItem) => countryItem.properties.country)
+  const allSpecies = []
+  const countrySelectedSpecies = []
+  let countrySelectedSpeciesWithAllSpecies = []
```

Comment:
I think you can avoid this being mutable by doing this on line 28. This is a nitpick though

```
const countrySelectedSpeciesWithAllSpecies = [
    ...uniqueCountrySelectedSpecies,
    ...filteredUniqueAllSpecies
  ]
  ```

Reply:
ohh yes good poiny